src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java

```diff
@@ -0,0 +1,735 @@
+//        this.observationUnitByNameNoScope = initialize_observationUnitByNameNoScope( program, experimentImportRows );
+        this.existingGermplasmByGID = initialize_existingGermplasmByGID( program, experimentImportRows );
+    }
+
+    /**
+     * @param importRows - one element of the list for every row of the import file.
+     * @param mappedBrAPIImport - passed in by reference and modified within this program (this will latter be passed to the front end for the preview)
+     * @param program
+     * @param user
+     * @param commit -  true when the data should be saved (ie when the user has pressed the "Commit" button)
+     *                  false when used for preview only
+     * @return Map<String, ImportPreviewStatistics> - used to display the summary statistics.
+     * @throws ValidatorException
+     */
+    @Override
+    public Map<String, ImportPreviewStatistics> process(
+            List<BrAPIImport> importRows,
+            Map<Integer, PendingImport> mappedBrAPIImport,
+            Program program,
+            User user,
+            boolean commit) throws ValidatorException {
+
+        ValidationErrors validationErrors = new ValidationErrors();
+
+        // add "New" pending data to the BrapiData objects
+        getNewBrapiData(importRows, program, commit);
+
+        // For each import row
+        for (int i = 0; i < importRows.size(); i++) {
+            ExperimentObservation importRow = (ExperimentObservation) importRows.get(i);
+
+            PendingImport mappedImportRow = mappedBrAPIImport.getOrDefault(i, new PendingImport());
+            mappedImportRow.setTrial( this.trialByNameNoScope.get( importRow.getExpTitle() ) );
+            mappedImportRow.setLocation( this.locationByName.get( importRow.getEnvLocation() ) );
+            mappedImportRow.setStudy( this.studyByNameNoScope.get( importRow.getEnv() ) );
+            mappedImportRow.setObservationUnit( this.observationUnitByNameNoScope.get( createOUKey( importRow ) ) );
+            PendingImportObject<BrAPIGermplasm> germplasmPIO = getGidPio(importRow);
+            mappedImportRow.setGermplasm( germplasmPIO );
+
+            if (! StringUtils.isBlank( importRow.getGid() )) { // if GID is blank, don't bother to check if it is valid.
+                validateGermplasm(importRow,validationErrors, i, germplasmPIO);
+            }
+            // Construct Observations -- Done in another card
+            mappedBrAPIImport.put(i, mappedImportRow);
+        }
+        // End-of-loop
+
+        validationErrors = validateFields(importRows, validationErrors);
+
+        if (validationErrors.hasErrors() ){
+            throw new ValidatorException(validationErrors);
+        }
+
+        // Construct our response object
+        return getStatisticsMap(importRows);
+    }
+
+    private void getNewBrapiData(List<BrAPIImport> importRows, Program program, boolean commit) {
+
+        String opsUnitSequenceName = program.getObsUnitSequence();
```

Comment:
should this be o**b**sUnitSequenceName?

Reply:
Yes.  I will change the name to obsUnitSequenceName.